checkov/common/util/prompt.py

```diff
@@ -0,0 +1,271 @@
+                                                            "default": "azurerm_policy_definition"
+                                                        },
+                                                    }
+                                                ]
+                                            }
+                                        },
+                                        {  # GCP
+                                            "context": {
+                                                "prompt_if": 'gcp',
+                                                "text": 'Select a terraform object for this check',
+                                                "type": click.Choice(['resource']),
+                                                "default": "resource",
+                                                "sub_prompts": [
+                                                    {
+                                                        "supported_resource": {
+                                                            "prompt_if": 'resource',
+                                                            "text": 'Enter the terraform object type',
+                                                            "type": str,
+                                                            "default": "google_project_iam_policy"
+                                                        },
+                                                    }
+                                                ]
+                                            }
+                                        },
+                                    ]
+                                }
+                            }
+                        ]
+                    }
+                }
+            ]
+        }
+    }
+
+    def __init__(self):
+        print(checkov_banner)
+        self.responses = {}
+        self.prompt()
+
+    # Recurse over our prompt, populating new class attributes from the keys and
+    # user-supplied answers
+    def prompt(self, prompt_map=PROMPTS, prompt_if=None):
+        for k, v in prompt_map.items():
+            if "prompt_if" not in v or v["prompt_if"] == prompt_if:
+                # Prompt the user
+                p = click.prompt(v["text"], type=v["type"], default=v["default"])
+
+                # Create the action on our object
+                if k == 'chosen_action':
+                    self.chosen_action = p
+
+                # Record user responses
+                self.responses[k] = p
+                print()  # Newline for readability
+
+                # Call prompt() again on any sub_prompts
+                if "sub_prompts" in v:
+                    for sub_prompt in v["sub_prompts"]:
+                        self.prompt(sub_prompt, prompt_if=p)
+
+    def template_env(self):
+        template_loader = jinja2.FileSystemLoader(searchpath=TEMPLATE_DIRECTORY)
+        return jinja2.Environment(loader=template_loader)
```

Comment:
@jmeredith16 got some bandit errors:
```
Issue: [B701:jinja2_autoescape_false] By default, jinja2 sets autoescape to False. Consider using autoescape=True or use the select_autoescape function to mitigate XSS vulnerabilities.
   Severity: High   Confidence: High
   Location: checkov/common/util/prompt.py:167
   More Info: https://bandit.readthedocs.io/en/latest/plugins/b701_jinja2_autoescape_false.html
``` 
can we turn autoscape here ?

Reply:
I tested it, there was no issue turning it to `True`
```suggestion
        return jinja2.Environment(loader=template_loader, autoescape=True)
```

Reply:
Done, sorry - meant to push that earlier this week